src/controllers/swapAndBridge/swapAndBridge.ts

```diff
@@ -411,11 +411,23 @@ export class SwapAndBridgeController extends EventEmitter {
     if (shouldEmit) this.emitUpdate()
   }
 
-  updatePortfolioTokenList(portfolioTokenList: TokenResult[]) {
-    this.portfolioTokenList = portfolioTokenList
+  updatePortfolioTokenList(nextPortfolioTokenList: TokenResult[]) {
+    this.portfolioTokenList = nextPortfolioTokenList
 
-    if (!this.fromSelectedToken) {
-      this.updateForm({ fromSelectedToken: this.portfolioTokenList[0] || null })
+    const fromSelectedTokenInNextPortfolio = nextPortfolioTokenList.find(
+      (t) => t.address === this.fromSelectedToken?.address
```

Comment:
Search by address and network combined

Reply:
Thanks, I just spot that. Fixed in https://github.com/AmbireTech/ambire-common/pull/1076/commits/1bf584a2f2dbfc168943d30ba9f7b085dd8bca07